asapdiscovery-data/asapdiscovery/data/schema_v2/complex.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any
+
+from asapdiscovery.data.openeye import load_openeye_pdb
+from asapdiscovery.data.schema_v2.ligand import Ligand
+from asapdiscovery.data.schema_v2.schema_base import DataModelAbstractBase
+from asapdiscovery.data.schema_v2.target import Target
+from asapdiscovery.modeling.modeling import split_openeye_mol
+from asapdiscovery.modeling.schema import MoleculeFilter
+from pydantic import Field
+
+
+class Complex(DataModelAbstractBase):
+    """
+    Schema for a Complex, containing both a Target and Ligand
+    """
+
+    target: Target = Field(description="Target schema object")
+    ligand: Ligand = Field(description="Ligand schema object")
+
+    # Overload from base class to check target and ligand individually
+    def data_equal(self, other: Complex):
+        return self.target.data_equal(other.target) and self.ligand.data_equal(
+            other.ligand
+        )
+
+    @classmethod
+    def from_pdb(
+        cls,
+        pdb_file: str | Path,
+        target_chains=[],
+        ligand_chain="",
+        target_kwargs={},
+        ligand_kwargs={},
+    ):
+        # First load full complex molecule
+        complex_mol = load_openeye_pdb(pdb_file)
+
+        # Split molecule into parts using given chains
+        mol_filter = MoleculeFilter(
+            protein_chains=target_chains, ligand_chain=ligand_chain
+        )
+        split_dict = split_openeye_mol(complex_mol, mol_filter)
```

Comment:
Add similar code path to `Target.from_pdb` to force there to be no ligands in a regular target as well.